lib/Sema/NameBinding.cpp

```diff
@@ -36,60 +37,539 @@
 using namespace swift;
 
 //===----------------------------------------------------------------------===//
-// NameBinder
+// MARK: NameBinder and supporting types
 //===----------------------------------------------------------------------===//
 
 using ImportedModule = ModuleDecl::ImportedModule;
+using ImportedModuleDesc = SourceFile::ImportedModuleDesc;
 using ImportOptions = SourceFile::ImportOptions;
+using ImportFlags = SourceFile::ImportFlags;
+
+namespace {
+  /// Represents an import which the NameBinder knows exists, but which has not
+  /// yet had its options checked, module loaded, or cross-imports found.
+  ///
+  /// An UnboundImport may represent a physical ImportDecl written in the
+  /// source, or it may represent a cross-import overlay that has been found and
+  /// needs to be loaded.
+  struct UnboundImport {
+    /// If this UnboundImport directly represents an ImportDecl, the ImportDecl
+    /// it represents.
+    ///
+    /// This property should only be used to control where information is added
+    /// to the AST. Don't retrieve information directly from the ImportDecl; use
+    /// the other member variables.
+    NullablePtr<ImportDecl> ID;
+
+    SourceLoc importLoc;
```

Comment:
Nitpick: Field names should be begin with uppercase letters. (Similarly in `BoundImport` and `NameBinder`).

Reply:
We've actually been transitioning away from this style in general. I made an exception for `ID` because `id` for an "import decl" just seems weird, but I'm thinking about changing this anyway.